sv.h

```diff
@@ -1051,6 +1051,17 @@ Remove any string offset.
 #define SvPOK_byte_pure_nogthink(sv) \
     ((SvFLAGS(sv) & (SVf_POK|SVf_UTF8|SVf_IOK|SVf_NOK|SVf_ROK|SVpgv_GP|SVf_THINKFIRST|SVs_GMG)) == SVf_POK)
 
+/*
+=for apidoc Am|BOOL|SvISBOOL|SV* sv
+
+Returns true if the SV is one of the special boolean constants (PL_sv_yes or
+PL_sv_no), or is a regular SV whose last assignment stored a copy of one.
+*/
+
+#define SvISBOOL(sv)                            \
+    (SvIOK(sv) && SvPOK(sv) && SvIsCOW_static(sv) && \
+        (SvPVX(sv) == (char *)&PL_Yes || SvPVX(sv) == (char *)&PL_No))
```

Comment:
Why take a pointer to the array (vs the default conversion to a pointer to the first character) and cast it?
```
(SvPVX(sv) == PL_Yes || SvPVX(sv) == PL_No))
```
would work too.

I thought it might be an attempt to avoid the -Wstring-compare warning from clang, but I don't see the warning if I use the simpler form.

Reply:
Oh that'll be because I can never remember exactly how those rules work, so I was probably being overly-cautious. I'll switch it to your simpler suggestion.